diff --git a/rclcpp_components/include/rclcpp_components/component_manager.hpp b/rclcpp_components/include/rclcpp_components/component_manager.hpp
index 62382842..15f68c92 100644
--- a/rclcpp_components/include/rclcpp_components/component_manager.hpp
+++ b/rclcpp_components/include/rclcpp_components/component_manager.hpp
@@ -109,19 +109,6 @@ public:
   RCLCPP_COMPONENTS_PUBLIC
   virtual ~ComponentManager();
 
-  /// Return a list of valid loadable components in a given package.
-  /**
-   * \param package_name name of the package
-   * \param resource_index name of the executable
-   * \throws ComponentManagerException if the resource was not found or a invalid resource entry
-   * \return a list of component resources
-   */
-  RCLCPP_COMPONENTS_PUBLIC
-  virtual std::vector<ComponentResource>
-  get_component_resources(
-    const std::string & package_name,
-    const std::string & resource_index = "rclcpp_components") const;
-
   /// Instantiate a component from a dynamic library.
   /**
    * \param resource a component resource (class name + library path)
diff --git a/rclcpp_components/src/component_manager.cpp b/rclcpp_components/src/component_manager.cpp
index 16329a42..82940c7d 100644
--- a/rclcpp_components/src/component_manager.cpp
+++ b/rclcpp_components/src/component_manager.cpp
@@ -76,35 +76,7 @@ ComponentManager::~ComponentManager()
   }
 }
 
-std::vector<ComponentManager::ComponentResource>
-ComponentManager::get_component_resources(
-  const std::string & package_name, const std::string & resource_index) const
-{
-  std::string content;
-  std::string base_path;
-  if (
-    !ament_index_cpp::get_resource(
-      resource_index, package_name, content, &base_path))
-  {
-    throw ComponentManagerException("Could not find requested resource in ament index");
-  }
 
-  std::vector<ComponentResource> resources;
-  std::vector<std::string> lines = rcpputils::split(content, '\n', true);
-  for (const auto & line : lines) {
-    std::vector<std::string> parts = rcpputils::split(line, ';');
-    if (parts.size() != 2) {
-      throw ComponentManagerException("Invalid resource entry");
-    }
-
-    std::string library_path = parts[1];
-    if (!rcpputils::fs::path(library_path).is_absolute()) {
-      library_path = base_path + "/" + library_path;
-    }
-    resources.push_back({parts[0], library_path});
-  }
-  return resources;
-}
 
 std::shared_ptr<rclcpp_components::NodeFactory>
 ComponentManager::create_component_factory(const ComponentResource & resource)
@@ -225,7 +197,10 @@ ComponentManager::on_load_node(
   (void) request_header;
 
   try {
-    auto resources = get_component_resources(request->package_name);
+    if (!rcpputils::fs::exists(request->package_name)) {
+      throw ComponentManagerException("Could not find requested resource in ament index");
+    }
+    std::vector<ComponentResource> resources = {{request->plugin_name, request->package_name}};
 
     for (const auto & resource : resources) {
       if (resource.first != request->plugin_name) {
