diff --git a/src/icp/icp_loop.cpp b/src/icp/icp_loop.cpp
index 4b1b7fd..c449f07 100644
--- a/src/icp/icp_loop.cpp
+++ b/src/icp/icp_loop.cpp
@@ -46,8 +46,8 @@ int icp_loop(struct sm_params*params, const double*q0, double*x_new,
 
 		/** If debug_verify_tricks, make sure that find_correspondences_tricks()
 		    and find_correspondences() return the same answer */
-			if(params->debug_verify_tricks)
-				debug_correspondences(params);
+		if(params->debug_verify_tricks)
+			debug_correspondences(params);
 
 		/* If not many correspondences, bail out */
 		int num_corr = ld_num_valid_correspondences(laser_sens);
diff --git a/src/icp/icp_outliers.cpp b/src/icp/icp_outliers.cpp
index bc68bab..b8034ca 100644
--- a/src/icp/icp_outliers.cpp
+++ b/src/icp/icp_outliers.cpp
@@ -21,7 +21,7 @@ void visibilityTest(LDP laser_ref, const gsl_vector*u) {
 	}
 	
 	sm_debug("\tvisibility: Found outliers: ");
-	int invalid = 0;
+	[[maybe_unused]] int invalid = 0;
 	for(j=1;j<laser_ref->nrays;j++) {
 		if(!ld_valid_ray(laser_ref,j)||!ld_valid_ray(laser_ref,j-1)) continue;
 		if(theta_from_u[j]<theta_from_u[j-1]) {
