diff --git a/orocos_kdl/src/chaindynparam.hpp b/orocos_kdl/src/chaindynparam.hpp
index 38fd225..b9dcbd7 100644
--- a/orocos_kdl/src/chaindynparam.hpp
+++ b/orocos_kdl/src/chaindynparam.hpp
@@ -59,7 +59,7 @@ namespace KDL {
 
     private:
         const Chain& chain;
-	int nr;  // unused, remove in a future version
+	[[maybe_unused]] int nr;  // unused, remove in a future version
 	unsigned int nj;
         unsigned int ns;	
 	Vector grav;
