diff --git a/rosbag2_cpp/src/rosbag2_cpp/cache/circular_message_cache.cpp b/rosbag2_cpp/src/rosbag2_cpp/cache/circular_message_cache.cpp
index d26d04e..b3e6c58 100644
--- a/rosbag2_cpp/src/rosbag2_cpp/cache/circular_message_cache.cpp
+++ b/rosbag2_cpp/src/rosbag2_cpp/cache/circular_message_cache.cpp
@@ -46,12 +46,14 @@ void CircularMessageCache::push(std::shared_ptr<const rosbag2_storage::Serialize
   producer_buffer_->push(msg);
 }
 
+__attribute__((no_thread_safety_analysis))
 std::shared_ptr<CacheBufferInterface> CircularMessageCache::get_consumer_buffer()
 {
   consumer_buffer_mutex_.lock();
   return consumer_buffer_;
 }
 
+__attribute__((no_thread_safety_analysis))
 void CircularMessageCache::release_consumer_buffer()
 {
   consumer_buffer_mutex_.unlock();
diff --git a/rosbag2_cpp/src/rosbag2_cpp/cache/message_cache.cpp b/rosbag2_cpp/src/rosbag2_cpp/cache/message_cache.cpp
index 139cb6f..a65ae06 100644
--- a/rosbag2_cpp/src/rosbag2_cpp/cache/message_cache.cpp
+++ b/rosbag2_cpp/src/rosbag2_cpp/cache/message_cache.cpp
@@ -60,12 +60,14 @@ void MessageCache::push(std::shared_ptr<const rosbag2_storage::SerializedBagMess
   notify_data_ready();
 }
 
+__attribute__((no_thread_safety_analysis))
 std::shared_ptr<CacheBufferInterface> MessageCache::get_consumer_buffer()
 {
   consumer_buffer_mutex_.lock();
   return consumer_buffer_;
 }
 
+__attribute__((no_thread_safety_analysis))
 void MessageCache::release_consumer_buffer()
 {
   consumer_buffer_mutex_.unlock();
diff --git a/rosbag2_cpp/src/rosbag2_cpp/serialization_format_converter_factory_impl.hpp b/rosbag2_cpp/src/rosbag2_cpp/serialization_format_converter_factory_impl.hpp
index b338a15..149957a 100644
--- a/rosbag2_cpp/src/rosbag2_cpp/serialization_format_converter_factory_impl.hpp
+++ b/rosbag2_cpp/src/rosbag2_cpp/serialization_format_converter_factory_impl.hpp
@@ -57,8 +57,7 @@ public:
         "rosbag2_cpp",
         converter_interfaces::SerializationFormatDeserializer::get_base_class_name());
     } catch (const std::exception & e) {
-      ROSBAG2_CPP_LOG_ERROR_STREAM("Unable to create class loader instance: " << e.what());
-      throw e;
+      ROSBAG2_CPP_LOG_DEBUG_STREAM("Unable to create class loader instance: " << e.what());
     }
   }
 
@@ -67,17 +66,26 @@ public:
   std::unique_ptr<converter_interfaces::SerializationFormatDeserializer>
   load_deserializer(const std::string & format)
   {
+    if (deserializer_class_loader_ == nullptr) {
+      return nullptr;
+    }
     return load_interface(format, deserializer_class_loader_);
   }
 
   std::unique_ptr<converter_interfaces::SerializationFormatSerializer>
   load_serializer(const std::string & format)
   {
+    if (serializer_class_loader_ == nullptr) {
+      return nullptr;
+    }
     return load_interface(format, serializer_class_loader_);
   }
 
   std::vector<std::string> get_declared_serialization_plugins() const
   {
+    if (serializer_class_loader_ == nullptr) {
+      return {};
+    }
     return serializer_class_loader_->getDeclaredClasses();
   }
 
diff --git a/rosbag2_py/src/rosbag2_py/_writer.cpp b/rosbag2_py/src/rosbag2_py/_writer.cpp
index bdac2c1..04242c9 100644
--- a/rosbag2_py/src/rosbag2_py/_writer.cpp
+++ b/rosbag2_py/src/rosbag2_py/_writer.cpp
@@ -83,12 +83,16 @@ std::unordered_set<std::string> get_registered_compressors()
 
 std::unordered_set<std::string> get_registered_serializers()
 {
-  auto serializers = rosbag2_cpp::plugins::get_class_plugins<
-    rosbag2_cpp::converter_interfaces::SerializationFormatSerializer>();
-  auto converters = rosbag2_cpp::plugins::get_class_plugins<
-    rosbag2_cpp::converter_interfaces::SerializationFormatConverter>();
-  serializers.insert(converters.begin(), converters.end());
-  return serializers;
+  try {
+    auto serializers = rosbag2_cpp::plugins::get_class_plugins<
+      rosbag2_cpp::converter_interfaces::SerializationFormatSerializer>();
+    auto converters = rosbag2_cpp::plugins::get_class_plugins<
+      rosbag2_cpp::converter_interfaces::SerializationFormatConverter>();
+    serializers.insert(converters.begin(), converters.end());
+    return serializers;
+  } catch (const std::exception&) {
+    return {};
+  }
 }
 
 }  // namespace rosbag2_py
diff --git a/rosbag2_transport/src/rosbag2_transport/player.cpp b/rosbag2_transport/src/rosbag2_transport/player.cpp
index 522d04c..7dffc44 100644
--- a/rosbag2_transport/src/rosbag2_transport/player.cpp
+++ b/rosbag2_transport/src/rosbag2_transport/player.cpp
@@ -323,6 +323,16 @@ private:
     int(
       const rosbag2_storage::SerializedBagMessageSharedPtr &,
       const rosbag2_storage::SerializedBagMessageSharedPtr &)>;
+  /// Comparator for SerializedBagMessageSharedPtr to order chronologically by recv_timestamp.
+  struct
+  {
+    bool operator()(
+      const rosbag2_storage::SerializedBagMessageSharedPtr & l,
+      const rosbag2_storage::SerializedBagMessageSharedPtr & r) const
+    {
+      return l->recv_timestamp > r->recv_timestamp;
+    }
+  } bag_message_chronological_recv_timestamp_comparator;
   LockedPriorityQueue<
     rosbag2_storage::SerializedBagMessageSharedPtr,
     std::vector<rosbag2_storage::SerializedBagMessageSharedPtr>,
@@ -371,17 +381,6 @@ private:
   std::vector<KeyboardHandler::callback_handle_t> keyboard_callbacks_;
 
   std::shared_ptr<PlayerServiceClientManager> player_service_client_manager_;
-
-  /// Comparator for SerializedBagMessageSharedPtr to order chronologically by recv_timestamp.
-  struct
-  {
-    bool operator()(
-      const rosbag2_storage::SerializedBagMessageSharedPtr & l,
-      const rosbag2_storage::SerializedBagMessageSharedPtr & r) const
-    {
-      return l->recv_timestamp > r->recv_timestamp;
-    }
-  } bag_message_chronological_recv_timestamp_comparator;
 };
 
 PlayerImpl::PlayerImpl(
